v2.1/online-schema-changes.md

```diff
@@ -0,0 +1,205 @@
+{{site.data.alerts.end}}
+
+{{site.data.alerts.callout_info}}
+Schema changes keep your data consistent at all times, but they do not run inside user transactions in the general case. This is necessary so the cluster can remain online and continue to service application reads and writes. For more information, see [Limitations](#limitations).
+{{site.data.alerts.end}}
+
+<span class="version-tag">New in v2.1:</span> You can run schema changes inside the same transaction as a [`CREATE TABLE`][create-table] statement. For more information, [see this example](#run-schema-changes-inside-a-transaction-with-create-table).
+
+## How online schema changes work
+
+At a high level, online schema changes are accomplished by using a bridging strategy involving concurrent uses of multiple versions of the schema. Schema changes are run as a series of transactions:
+
+1. A user-level transaction initiated by [`ALTER TABLE`][alter-table], [`CREATE INDEX`][create-index], [`TRUNCATE`][truncate], etc. The user-level transaction schedules the schema change for the schema change engine to complete.
+
+2. A series of system-level transactions calculated by the schema change engine that convert the original schema to the new schema in discrete steps while ensuring that the underlying table data is always in a consistent state. These transactions are executed as a [background job][show-jobs].
+
+This approach allows the schema change engine to roll out a new schema while the previous version is still in use. It then backfills or deletes the underlying table data as needed in the background, while the cluster is still running and servicing reads and writes from your application.
+
+During the backfilling process, the schema change engine updates the underlying table data to make sure all instances of the table are stored according to the requirements of the new schema.
+
+Once backfilling is complete, all nodes will switch over to the new schema, and will allow reads and writes of the table using the new schema.
+
+For more technical details, see [How online schema changes are possible in CockroachDB][blog].
+
+## Limitations
+
+Schema changes cannot be performed inside user [transactions][txns] in the general case. Instead, they run as background jobs (that can be viewed with [`SHOW JOBS`][show-jobs]). This behavior is necessary because making schema changes transactional would mean requiring a given schema change transaction to propagate across all the nodes of a cluster. This would block all user-initiated transactions being run by your application, since the schema-change-containing transaction would have to commit before any other transactions could make progress. This would prevent the cluster from servicing reads and writes during the schema change, requiring application downtime.
+
+For example, this means that the following queries will result in errors.
+
+Create an index and then run a select against that index inside a user-level transaction:
```

Comment:
I'd cut `user-level` throughout. It's not clear what it's in contrast to in this (or other docs).

Reply:
Fixed.  I'm going with your proposed strategy below to avoid mentioning the distinction between "user-level" and "system" transactions since, as you note, it is not really mentioned anywhere else, and it's probably too much detail.